generators/app/templates/elm/document/src/Main.elm

```diff
@@ -0,0 +1,122 @@
+module Main exposing (Model(..), Msg(..), getRandomCatGif, gifDecoder, init, main, subscriptions, update, view, viewGif)
+
+import Browser
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (..)
+import Http
+import Json.Decode exposing (Decoder, field, string)
+
+
+
+-- MAIN
+
+
+main =
+    Browser.document
+        { init = init
+        , update = update
+        , subscriptions = subscriptions
+        , view = viewDocument
+        }
+
+
+
+-- MODEL
+
+
+type Model
+    = Failure
+    | Loading
+    | Success String
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( Loading, getRandomCatGif )
+
+
+
+-- UPDATE
+
+
+type Msg
+    = MorePlease
+    | GotGif (Result Http.Error String)
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        MorePlease ->
+            ( Loading, getRandomCatGif )
+
+        GotGif result ->
+            case result of
+                Ok url ->
+                    ( Success url, Cmd.none )
+
+                Err _ ->
+                    ( Failure, Cmd.none )
+
+
+
+-- SUBSCRIPTIONS
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Sub.none
+
+
+
+-- VIEW
+
+
+viewDocument : Model -> Browser.Document Msg
+viewDocument model =
+    { title = "Some cats", body = [ view model ] }
+
+
+view : Model -> Html Msg
+view model =
+    div []
+        [ h2 [] [ text "Random Cats" ]
+        , viewGif model
+        ]
+
+
+viewGif : Model -> Html Msg
+viewGif model =
+    case model of
+        Failure ->
+            div []
+                [ text "I could not load a random cat for some reason. "
+                , button [ onClick MorePlease ] [ text "Try Again!" ]
+                ]
+
+        Loading ->
+            text "Loading..."
+
+        Success url ->
+            div []
+                [ button [ onClick MorePlease, style "display" "block" ] [ text "More Please!" ]
+                , img [ src url ] []
+                ]
+
+
+
+-- HTTP
+
+
+getRandomCatGif : Cmd Msg
+getRandomCatGif =
+    Http.get
+        { url = "https://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC&tag=cat"
```

Comment:
I think I'd rather give users an app that _probably_ works rather than need them to go and get an API key to see the example run. I'm good with this as is, I guess. Just wanted to think about it a bit.